github.go

```diff
@@ -61,8 +66,24 @@ func getCommits(issueable Issueable, pullRequests PullRequests) ([]github.Reposi
 func addLabel(repository Repository, issueNumber int, label string, issues Issues) *ErrorResponse {
 	_, _, err := issues.AddLabelsToIssue(repository.Owner, repository.Name, issueNumber, []string{label})
 	if err != nil {
-		message := fmt.Sprintf("Failed to set the label %s for issue nr %d", label, issueNumber)
+		message := fmt.Sprintf("Failed to set the label %s for issue #%d", label, issueNumber)
 		return &ErrorResponse{err, http.StatusBadGateway, message}
 	}
 	return nil
 }
+
+func merge(repository Repository, issueNumber int, pullRequests PullRequests) error {
+	additionalCommitMessage := ""
+	result, resp, err := pullRequests.Merge(repository.Owner, repository.Name, issueNumber, additionalCommitMessage)
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusMethodNotAllowed {
+			return NotMergeableError
+		} else if resp != nil && resp.StatusCode == http.StatusConflict {
+			return OutdatedMergeRefError
+		}
+		return err
+	} else if result.Merged == nil || !*result.Merged {
+		return errors.New("Request successful, but PR not merged.")
```

Comment:
Not entirely clear from reading the docs. So currently just considering this as an error, maybe later when testing against live GitHub if we understand this scenario better then we can implement some recovery code here.